examples/tpm2-seal-unseal/main.go

```diff
@@ -0,0 +1,281 @@
+	}
+	for otherErr := range otherErrs {
+		errs = append(errs, fmt.Errorf("error while unsealing data: %v", otherErr))
+	}
+	if len(errs) > 0 {
+		return
+	}
+	// Note that this can be easily fooled since we don't verify that the error was
+	// due specifically to policy.
+	fmt.Printf("As expected, unsealing after extending the PCR failed with an error. Error was: %v\n", unsealErr)
+
+	return
+}
+
+// Returns the unsealed data, any error due to the Unseal command, and any other errors
+func unseal(rwc io.ReadWriteCloser, pcr int, objectPassword string, objectHandle tpmutil.Handle) (data []byte, unsealErr error, otherErrs []error) {
+	// Create the authorization session
+	sessHandle, _, err := policyPCRPasswordSession(rwc, pcr, objectPassword)
+	defer func() {
+		if sessHandle != tpm2.HandleNull {
+			if flushErr := tpm2.FlushContext(rwc, sessHandle); flushErr != nil {
+				otherErrs = append(otherErrs, fmt.Errorf("unable to flush session: %v", flushErr))
+			}
+		}
+	}()
+	if err != nil {
+		otherErrs = append(otherErrs, fmt.Errorf("unable to get auth session: %v", err))
+		return
+	}
+
+	// Unseal the data
+	unsealedData, err := tpm2.UnsealWithSession(rwc, sessHandle, objectHandle, objectPassword)
+	if err != nil {
+		unsealErr = fmt.Errorf("unable to unseal data: %v", err)
+		return
+	}
+	data = unsealedData
+	return
+}
+
+// Note that the handle may be returned even if an error is returned as well, in which
+// case it still needs to be flushed.
+// Returns session handle and policy digest.
+func policyPCRPasswordSession(rwc io.ReadWriteCloser, pcr int, password string) (tpmutil.Handle, []byte, error) {
+	// FYI, this is not a very secure session.
+	sessHandle, _, err := tpm2.StartAuthSession(
+		rwc,
+		tpm2.HandleNull,                                        /*tpmKey*/
+		tpm2.HandleNull,                                        /*bindKey*/
+		[]byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}, /*nonceCaller*/
+		nil, /*secret*/
+		tpm2.SessionPolicy,
+		tpm2.AlgNull,
+		tpm2.AlgSHA256)
+	if err != nil {
+		return tpm2.HandleNull, nil, fmt.Errorf("unable to start session: %v", err)
+	}
+
+	pcrSelection := tpm2.PCRSelection{
+		Hash: tpm2.AlgSHA256,
+		PCRs: []int{pcr},
+	}
+
+	// An empty expected digest means that digest verification is skipped.
+	err = tpm2.PolicyPCR(rwc, sessHandle, nil /*expectedDigest*/, pcrSelection)
+	if err != nil {
+		return sessHandle, nil, fmt.Errorf("unable to bind PCRs to auth policy: %v", err)
+	}
+
+	err = tpm2.PolicyPassword(rwc, sessHandle)
+	if err != nil {
+		return sessHandle, nil, fmt.Errorf("unable to require password for auth policy: %v", err)
+	}
+
+	policy, err := tpm2.PolicyGetDigest(rwc, sessHandle)
+	if err != nil {
+		return sessHandle, nil, fmt.Errorf("unable to get policy digest: %v", err)
+	}
+	return sessHandle, policy, nil
+}
+
+func readPCR(rwc io.ReadWriteCloser, pcr int) ([]byte, error) {
```

Comment:
I wonder if we should add this helper func to tpm2 directly. seems useful

Reply:
sure, np, I'll send a separate PR